packages/strapi-parts/src/Combobox/Combobox.js

```diff
@@ -0,0 +1,274 @@
+      'aria-selected': selectedIndex === i ? 'true' : false,
+      ref: (r) => {
+        if (isActive) activeOptionRef.current = r;
+      },
+      onClick: () => onOptionClick(i),
+      onMouseDown: onOptionMouseDown,
+      isSelected: isActive,
+    });
+  });
+
+  return (
+    <>
+      <MainRow ref={containerRef} disabled={disabled}>
+        <Input
+          aria-activedescendant={activeId}
+          aria-autocomplete="list"
+          aria-controls={`${htmlId}-listbox`}
+          aria-expanded={`${open}`}
+          aria-haspopup="listbox"
+          aria-label={label}
+          aria-disabled={disabled}
+          readOnly={disabled}
+          ref={inputRef}
+          role="combobox"
+          type="text"
+          value={inputValue}
+          onBlur={!disabled ? onInputBlur : undefined}
+          onClick={!disabled ? () => updateMenuState(true) : undefined}
+          onInput={!disabled ? onInput : undefined}
+          onKeyDown={!disabled ? onInputKeyDown : undefined}
+          placeholder={placeholder}
+        />
+        <Row>
+          <CaretBox
+            disabled={disabled}
+            paddingLeft={3}
+            aria-hidden
+            as="button"
+            onClick={() => {
+              inputRef.current.focus();
+              updateMenuState(true);
+            }}
+            tabIndex={-1}
+          >
+            <DropdownIcon />
+          </CaretBox>
+        </Row>
+      </MainRow>
+      {open && (
+        <Popover
+          source={containerRef}
+          spacingTop={1}
+          fullWidth
+          intersectionId={`${htmlId}-listbox-popover-intersection`}
+          onReachEnd={hasMoreItems && !loading ? onLoadMore : undefined}
+        >
+          <div role="listbox" ref={listboxRef} id={`${htmlId}-listbox`}>
+            {Boolean(filteredNodes.length) ? (
+              filteredNodesClone
+            ) : creatable ? (
+              <OptionBox
+                paddingLeft={4}
+                paddingRight={4}
+                paddingTop={2}
+                paddingBottom={2}
+                ref={activeOptionRef}
+                onMouseDown={onOptionMouseDown}
+                role="option"
+                onClick={() => onOptionSelect()}
+              >
+                <Text textColor="neutral800">{createMessage(inputValue)}</Text>
+              </OptionBox>
+            ) : (
+              <Box paddingLeft={4} paddingRight={4} paddingTop={2} paddingBottom={2} ref={activeOptionRef}>
+                <Text textColor="neutral800">{noOptionsMessage(inputValue)}</Text>
+              </Box>
+            )}
+            {loading && (
+              <Row justifyContent="center" alignItems="center" paddingTop={2} paddingBottom={2}>
+                <Loader small>Loading content...</Loader>
+              </Row>
+            )}
+          </div>
+        </Popover>
+      )}
+    </>
+  );
+};
+
+Combobox.defaultProps = {
+  createMessage: (value) => `Create "${value}"`,
```

Comment:
Aren't this props and the `createable` one redundant? WYT of only using one prop since the message should always be provided when `createable` is true

Reply:
Right, but you could want the component to be creatable without providing a createMessage value.
Or maybe we could do like react-select, by having a <Creatable /> component

Reply:
The Creatable component makes sense for compasibility I think, do you feel you could do it that way?